ruby/ql/lib/codeql/ruby/frameworks/ActiveSupport.qll

```diff
@@ -60,6 +65,90 @@ module ActiveSupport {
       }
     }
 
+    /**
+     * Extensions to the `Object` class.
+     */
+    module Object {
+      /** Flow summary for methods which can return the receiver. */
+      private class IdentitySummary extends SimpleSummarizedCallable {
+        IdentitySummary() { this = ["presence", "deep_dup"] }
+
+        override predicate propagatesFlowExt(string input, string output, boolean preservesValue) {
+          input = "Argument[self]" and
+          output = "ReturnValue" and
+          preservesValue = true
+        }
+      }
+    }
+
+    /**
+     *    Extensions to the `Hash` class.
+     */
+    module Hash {
+      private class WithIndifferentAccessSummary extends SimpleSummarizedCallable {
+        WithIndifferentAccessSummary() { this = "with_indifferent_access" }
+
+        override predicate propagatesFlowExt(string input, string output, boolean preservesValue) {
+          input = "Argument[self].Element[any]" and
+          output = "ReturnValue.Element[any]" and
+          preservesValue = true
+        }
+      }
+
+      private class TransformSummary extends SimpleSummarizedCallable {
+        TransformSummary() {
+          this =
+            [
+              "stringify_keys", "to_options", "symbolize_keys", "deep_stringify_keys",
+              "deep_symbolize_keys", "with_indifferent_access"
+            ]
+        }
+
+        override predicate propagatesFlowExt(string input, string output, boolean preservesValue) {
+          input = "Argument[self].Element[any]" and
+          output = "ReturnValue.Element[?]" and
+          preservesValue = true
+        }
+      }
+
+      private string getExtractComponent(MethodCall mc, int i) {
+        mc.getMethodName() = "extract!" and
+        result = DataFlow::Content::getKnownElementIndex(mc.getArgument(i)).serialize()
+      }
+
+      private class ExtractSummary extends SummarizedCallable {
```

Comment:
This summary is sufficiently complicated that I can't quite follow what it's doing. Could you add a comment, perhaps with an example call to `extract!` and then an explanation of the flow step we're adding?